coap-core/src/main/java/com/mbed/coap/server/CoapServerBuilder.java

```diff
@@ -120,7 +120,7 @@ public CoapServerBuilder routeFilter(Filter.SimpleFilter<CoapRequest, CoapRespon
         return this;
     }
 
-    public CoapServerBuilder inboundRequestFilter(Filter.SimpleFilter<CoapRequest, CoapResponse> inboundRequestFilter) {
+    public CoapServerBuilder inboundRequestFilter(Filter<CoapRequest, CoapResponse, CoapRequest, CoapResponse> inboundRequestFilter) {
```

Comment:
What is the reason for this change?

Reply:
Without this I couldn't pass `filter.andThen(otherFilter)` to `CoapServerBuilder.inboundRequestFilter()`.

Reply:
Well, this can be worked around by just casting to `Filter.SimpleFilter` of the `andThen()` result.